### DISCUSSION TITLE
OpenShift Update Service for OKD

### DISCUSSION THREAD

**Discussion by disposab1e** [okd-project/okd#946] [Q&A]
Cause we're running OKD in an disconnected environment we're very interested in using the OpenShift Update Service as described [here](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.8/html/updating_clusters/installing-update-service).

Everything works fine but the missing bits are the cincinnati graph data for OKD. For OpenShift it is available [here](https://github.com/openshift/cincinnati-graph-data).

Any chance to find a repo like this for OKD too?

**Comment by vrutkovs** [comment]
OKD doesn't use OSUS instance, but uses Cincinnati protocol to control upgrade graph.

OSUS allows us to modify graph using channels and edge blocks via cincinnati-graph-data. OKD is using a simplified version of it - github.com/openshift/release-controller, which can build the graph dynamically based on Prow upgrade jobs (if the test passes the upgrade edge is added). Release controller channels are defined [in release repo](https://github.com/openshift/release/tree/master/core-services/release-controller/_releases), but has no concept of blocked edges or promotion windows.

However you can still use OSUS and OSUS operator to update a disconnected fleet of OKD instances - but you'd have to come up with settings and cincinnati-graph-data repo yourself.

**Comment by disposab1e** [comment]
@vrutkovs Thx for your detailed reply! I added a channel `stable-4` to cincinnati graph data repo with following content:

>  but you'd have to come up with settings and cincinnati-graph-data repo yourself

/channels/stable-4.yaml

```
name: stable-4
versions:
- 4.8.0-0.okd-2021-10-24-061736
- 4.8.0-0.okd-2021-11-14-052418
```

/blocked-edges/4.8.0-0.okd-2021-10-24-061736.yaml

```
to: 4.8.0-0.okd-2021-10-24-061736
from: 4\.7\..*
```
/blocked-edges/4.8.0-0.okd-2021-11-14-052418.yaml

```
to: 4.8.0-0.okd-2021-11-14-052418
from: 4\.7\..*
```



I don't want to waste your time. Just a quick question... is this enough? yes/no

Of course additionally I did everything documented [here](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.8/html/updating_clusters/installing-update-service).

**Reply by vrutkovs** [reply]
Looks sufficient to me

**Comment by disposab1e** [comment]
/channels/okd-4.yaml

```
name: okd-4
versions:
- 4.8.0-0.okd-2021-10-24-061736
- 4.8.0-0.okd-2021-11-14-052418
```

Everything looks good... but the next version is not available... @vrutkovs do you have a hint what can be missing?

<img width="833" alt="2021-11-19_07-50-45" src="https://user-images.githubusercontent.com/48065501/142578103-e6c9a3ce-0f4f-4067-bfa3-a82cd7585194.png">

**Comment by disposab1e** [comment]
@vrutkovs do you have a hint what is missing?
